src/language/dafnyInstallation.ts

```diff
@@ -159,6 +159,19 @@ export class DafnyInstaller {
       this.writeStatus('If you got `brew: command not found`, but brew is installed on your system, please add all brew commands to your ~/.zprofile, e.g. https://apple.stackexchange.com/a/430904 and reinstall Dafny.');
       return false;
     }
+    try {
+      const result = (await this.execLog('javac -version')).stdout;
+      if(!(/javac \d+\.\d+/.exec(result))) {
```

Comment:
What do we need java for?

Reply:
This is related to the issue I've opened last year: https://github.com/dafny-lang/dafny/issues/1530
At this time, the Dafny runtime requires java to compile, even though the C# distribution doesn't require it. I'd rather fix this issue than introducing another workaround.

Reply:
afaik the previous change was to enable M1 support by compiling Dafny from source. Therefore, the `javac` command is required because the `DafnyRuntime` project still relies on it.